webpack/assets/javascripts/react_app/routes/common/EmptyPage/RedirectedEmptyPage.js

```diff
@@ -0,0 +1,83 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Button } from '@patternfly/react-core';
+import { useHistory } from 'react-router-dom';
+import { SearchIcon } from '@patternfly/react-icons';
+import { translate as __ } from '../../../common/I18n';
+import { visit } from '../../../../foreman_navigation';
+import { EmptyStatePattern } from '../../../components/common/EmptyState';
+
+const StatefullEmptyPage = ({ location: { state = {} } }) => {
```

Comment:
The filename is `RedirectedEmptyPage`, so this name is confusing. Should we unify?